src/moq_instance.cpp

```diff
@@ -0,0 +1,1672 @@
+                conn_it->second.tracks_by_sub_id.clear();
+
+                // TODO(tievens): Clean up publish tracks
+
+#ifndef LIBQUICR_WITHOUT_INFLUXDB
+                _mexport.del_conn_ctx_info(conn_id);
+#endif
+                _connections.erase(conn_it);
+
+                break;
+            }
+            case TransportStatus::Shutdown:
+                _status = Status::NOT_READY;
+                break;
+        }
+    }
+
+    void MoQInstance::on_new_connection(const TransportConnId& conn_id, const TransportRemote& remote)
+    {
+        auto [conn_ctx, is_new] = _connections.try_emplace(conn_id, ConnectionContext{});
+
+        _logger->info << "New connection conn_id: " << conn_id
+                      << " remote ip: " << remote.host_or_ip
+                      << " port: " << remote.port
+                      << std::flush;
+
+        conn_ctx->second.conn_id = conn_id;
+    }
+
+    void MoQInstance::on_recv_stream(const TransportConnId& conn_id,
+                        uint64_t stream_id,
+                        std::optional<DataContextId> data_ctx_id,
+                        const bool is_bidir)
+    {
+        auto stream_buf = _transport->getStreamBuffer(conn_id, stream_id);
+
+        //std::lock_guard<std::mutex> _(_state_mutex);
+
+        auto& conn_ctx = _connections[conn_id];
+
+        if (stream_buf == nullptr) {
+            return;
+        }
+
+        if (is_bidir && not conn_ctx.ctrl_data_ctx_id) {
+            if (not data_ctx_id) {
+                close_connection(conn_id,
+                                 MoQTerminationReason::INTERNAL_ERROR,
+                                 "Received bidir is missing data context");
+                return;
+            }
+            conn_ctx.ctrl_data_ctx_id = data_ctx_id;
+        }
+
+        for (int i=0; i < 60; i++) { // don't loop forever, especially on bad stream
+            // bidir is Control stream, data streams are unidirectional
+            if (is_bidir) {
+                if (not conn_ctx.ctrl_msg_type_received) {
+                    auto msg_type = stream_buf->decode_uintV();
+
+                    if (msg_type) {
+                        conn_ctx.ctrl_msg_type_received = static_cast<MoQMessageType>(*msg_type);
+                    } else {
+                        break;
+                    }
+                }
+
+                if (conn_ctx.ctrl_msg_type_received) {
+                    if (process_recv_ctrl_message(conn_ctx, stream_buf)) {
+                        conn_ctx.ctrl_msg_type_received = std::nullopt;
+                        break;
+                    }
+                }
+            }
+
+            // Data stream, unidirectional
+            else {
+                if (process_recv_stream_data_message(conn_ctx, stream_buf)) {
+                    break;
+                }
+            }
+
+
+            if (!stream_buf->size()) { // done
+                break;
+            }
```

Comment:
I feel like some reordering of if statements and compressing of conditions might flatten the tabbing on this a bit... but the exact layout is escaping me, so this should be fine.

Reply:
It is a bit funky due to the stream processing of not always being able to get the uintvar.  Only if we actually get it, will it be set. Otherwise, it'll remain unset.